modelserving.mdx

```diff
@@ -0,0 +1,128 @@
+---
+title: 'Model Serving on GPUs'
+
+---
+
+Model Deployment is a very common GPU use case. With Shadeform, it's easy to deploy models right to the most affordable gpu's in the market with just a few commands. 
+
+In this guide, we will deploy [Mistral-7b-v0.1](https://huggingface.co/mistralai/Mistral-7B-v0.1) with VLLM onto an A6000.
+
+### Setup
+
+This guide builds off of our others for [finding the best gpu](https://docs.shadeform.ai/findingmostaffordablegpus) and for [deploying gpu containers](https://docs.shadeform.ai/deployinggpucontainers).
+We have a python notebook already to go for you to deploy this model that you can [find here](https://github.com/shadeform/examples/blob/main/basic_serving_vllm.ipynb).
+The requirements are simple, so in a python environment with (`requests` + optionally `openai`) installed:
+
+```bash
+git clone https://github.com/shadeform/examples.git
+cd examples/
+```
+Then in `basic_serving_vllm.ipynb` you will need to input your [Shadeform API Key](https://platform.shadeform.ai/settings/api). 
+
+### Serving a Model
+
+Once we have an instance, we deploy a model serving container with this request payload.
+
+```python
+model_id = "mistralai/Mistral-7B-v0.1"
+
+payload = {
+  "cloud": best_instance["cloud"],
+  "region": region,
+  "shade_instance_type": shade_instance_type,
+  "shade_cloud": True,
+  "name": "cool_gpu_server",
+  "launch_configuration": {
+    "type": "docker",
+     #This selects the image to launch, and sets environment variables "tasks" and "num_fewshot"
+    "docker_configuration": {
+      "image": "vllm/vllm-openai:latest",
+      "args": "--model " + model_id,
+      "envs": [],
+      "port_mappings": [
+        {
+          "container_port": 8000,
+          "host_port": 8000
+        }
+      ]
+    }
+  }
+}
+
+#request the best instance that is available
+response = requests.request("POST", create_url, json=payload, headers=headers)
+#easy way to visually see if this request worked
+print(response.text)
+```
+
+Once we request it, Shadeform will provision the machine, and deploy a docker container based on the image, arguments, and environment variables that we selected. 
+
+We can see that this will deploy an openai compatible server with VLLM serving Mistral-7b-v0.1.
```

Comment:
nit: Can we make the v in vLLM lowercase? That's their preferred way of writing it